main.nf

```diff
@@ -8,269 +8,4129 @@
+        summary['Path to mzML identification files (positive)'] = params.id_mzml_files_pos
+    }
+}
+
+if (params.type_of_ionization in (['neg', 'both'])) {
+    summary['Path to mzML quantification files (negative)'] = params.quant_mzml_files_neg
+    if (params.perform_identification == true) {
+        summary['Path to mzML identification files (negative)'] = params.id_mzml_files_neg
+    }
+}
+
+
+summary['Max Resources']    = "$params.max_memory memory, $params.max_cpus cpus, $params.max_time time per job"
+if (workflow.containerEngine) summary['Container'] = "$workflow.containerEngine - $workflow.container"
+summary['Output dir']       = params.outdir
+summary['Launch dir']       = workflow.launchDir
+summary['Working dir']      = workflow.workDir
+summary['Script dir']       = workflow.projectDir
+summary['User']             = workflow.userName
+if (workflow.profile.contains('awsbatch')) {
+    summary['AWS Region']   = params.awsregion
+    summary['AWS Queue']    = params.awsqueue
+    summary['AWS CLI']      = params.awscli
+}
+summary['Config Profile'] = workflow.profile
+if (params.config_profile_description) summary['Config Profile Description'] = params.config_profile_description
+if (params.config_profile_contact)     summary['Config Profile Contact']     = params.config_profile_contact
+if (params.config_profile_url)         summary['Config Profile URL']         = params.config_profile_url
+summary['Config Files'] = workflow.configFiles.join(', ')
+if (params.email || params.email_on_fail) {
+    summary['E-mail Address']    = params.email
+    summary['E-mail on failure'] = params.email_on_fail
+}
+
+// Check the hostnames against configured profiles
+checkHostname()
+
+Channel.from(summary.collect{ [it.key, it.value] })
+    .map { k,v -> "<dt>$k</dt><dd><samp>${v ?: '<span style=\"color:#999999;\">N/A</a>'}</samp></dd>" }
+    .reduce { a, b -> return [a, b].join("\n            ") }
+    .map { x -> """
+    id: 'nf-core-metaboigniter-summary'
+    description: " - this information is collected when the pipeline is started."
+    section_name: 'nf-core/metaboigniter Workflow Summary'
+    section_href: 'https://github.com/nf-core/metaboigniter'
+    plot_type: 'html'
+    data: |
+        <dl class=\"dl-horizontal\">
+            $x
+        </dl>
+    """.stripIndent() }
+    .set { ch_workflow_summary }
+
+/*
+ * Parse software version numbers
+ */
+process get_software_versions {
+    publishDir "${params.outdir}/pipeline_info", mode: params.publish_dir_mode,
+        saveAs: { filename ->
+                      if (filename.indexOf('.csv') > 0) filename
+                      else null
+        }
+
+    output:
+    file 'software_versions_mqc.yaml' into ch_software_versions_yaml
+    file 'software_versions.csv'
+
+
+    script:
+    // TODO nf-core: Get all tools to print their version number here fastqc --version > v_fastqc.txt
+    """
+    echo $workflow.manifest.version > v_pipeline.txt
```

Comment:
as mentioned above you could expand with other software such as OpenMS and co

Reply:
Each process has it's own container. So i should either manually do that come up with a workaround. But it's a good suggestion

Reply:
With DSL2 we are doing the version calls after the main step in the process, eg: [here](https://github.com/nf-core/modules/blob/983ba000c1d55297d8f32e29cac0a52229d1c757/software/rseqc/bamstat/main.nf#L37)

```nextflow
    output:
    tuple val(meta), path("*.my_output.txt"), emit: txt
    path  "*.version.txt"                  , emit: version

    script:
    def software = getSoftwareName(task.process)
    """
    my_command.py -i $bam $options.args > ${prefix}.my_output.txt
    my_command.py --version > ${software}.version.txt
    """
```

I think that it would be a good approach to do the same here, then scoop all of these channels up into the software_versions process. This way you get the versions and you're already part-way towards DSL2 style..

Reply:
Fixed using a single container.